ChangeSet
  1.1545 04/03/31 15:29:56 boutcher@skunk.rchland.ibm.com +1 -0
  Correctly handle long_busy return code from pSeries hypervisor in
  the ibmvscsi driver

  drivers/scsi/ibmvscsi/rpa_vscsi.c
    1.16 04/03/31 15:29:35 boutcher@skunk.rchland.ibm.com +4 -1
    Correctly handle long_busy return code from the hypervisor

diff -Nru a/drivers/scsi/ibmvscsi/rpa_vscsi.c b/drivers/scsi/ibmvscsi/rpa_vscsi.c
--- a/drivers/scsi/ibmvscsi/rpa_vscsi.c	Thu Apr  1 06:44:54 2004
+++ b/drivers/scsi/ibmvscsi/rpa_vscsi.c	Thu Apr  1 06:44:54 2004
@@ -69,9 +69,12 @@
 				struct ibmvscsi_host_data *hostdata,
 				int max_requests)
 {
+	long rc;
 	struct vio_dev *vdev = to_vio_dev(hostdata->dev);
 	free_irq(vdev->irq, (void *)hostdata);
-	plpar_hcall_norets(H_FREE_CRQ, vdev->unit_address);
+	do {
+		rc = plpar_hcall_norets(H_FREE_CRQ, vdev->unit_address);
+	} while (H_isLongBusy(rc));
 	dma_unmap_single(hostdata->dev,
 			 queue->msg_token,
 			 queue->size * sizeof(*queue->msgs),
.........................................................................
# vim: syntax=diff

